### ISSUE TITLE
Add support for more metrics

### ISSUE THREAD

**Issue by casper-hansen** [casperbh96/Nested-Cross-Validation#6]
Go through each metric from [sklearn.metrics](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) and make sure every metric works.

Supported metrics: the ones with `metric(y_true, y_pred)`

(i.e. roc_auc_score doesn't work, since it implements a `metric(y_true, y_score)`)

**Comment by casper-hansen** [comment]
Support for `predict_proba` on model must be implemented to get a `y_score.`

**Comment by casper-hansen** [comment]
Support added for binary `roc_auc_score` in dev

**Comment by casper-hansen** [comment]
Support added for multiclass scoring metrics. Use 'multiclass_average' in cv_options and specify either ‘micro’, ‘macro’, ‘samples’ or ‘weighted’
